panic if hstore has nil value

If hstore has nil value, Save() will panic.

```
package main

import (
	"fmt"

	_ "github.com/go-sql-driver/mysql"
	"github.com/jinzhu/gorm"
	_ "github.com/lib/pq"
	_ "github.com/mattn/go-sqlite3"
)

type Details struct {
	Id   int64
	Data gorm.Hstore
}

func main() {
	db, _ := gorm.Open("postgres", "user=ikuo dbname=test sslmode=disable")
	db.DB()

	db.Exec("CREATE EXTENSION IF NOT EXISTS hstore")
	db.DropTableIfExists(&Details{})
	db.CreateTable(&Details{})

	r := Details{
		Data: map[string]*string{
			"key": nil,
		},
	}
	if err := db.Save(&r).Error; err != nil {
		fmt.Println(err)
	}
}
```
The error is following below:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4153dbd]

goroutine 1 [running]:
github.com/jinzhu/gorm.Hstore.Value(0xc20803b350, 0x0, 0x0, 0x0, 0x0)
	/Users/ikuo/lib/go/src/github.com/jinzhu/gorm/postgres.go:109 +0x20d
database/sql/driver.defaultConverter.ConvertValue(0x43b5200, 0xc20803b350, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/database/sql/driver/types.go:219 +0x103
database/sql.driverArgs(0x0, 0xc2080b6370, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/database/sql/convert.go:35 +0x1dd
database/sql.(*DB).queryConn(0xc208046000, 0xc208044c60, 0x44a1ce8, 0xc20802db80, 0x43, 0xc2080b6370, 0x1, 0x1, 0x43, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/database/sql/sql.go:945 +0xcb
database/sql.(*Tx).Query(0xc20802db30, 0xc20802db80, 0x43, 0xc2080b6370, 0x1, 0x1, 0x44074f0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/database/sql/sql.go:1259 +0x10b
database/sql.(*Tx).QueryRow(0xc20802db30, 0xc20802db80, 0x43, 0xc2080b6370, 0x1, 0x1, 0x43)
	/usr/local/Cellar/go/1.4.2/libexec/src/database/sql/sql.go:1266 +0x66
github.com/jinzhu/gorm.Create(0xc208049e80)
	/Users/ikuo/lib/go/src/github.com/jinzhu/gorm/callback_create.go:73 +0x88c
github.com/jinzhu/gorm.(*Scope).callCallbacks(0xc208049e80, 0xc20803fb40, 0x8, 0x8, 0xc200000000)
	/Users/ikuo/lib/go/src/github.com/jinzhu/gorm/scope_private.go:297 +0x6b
github.com/jinzhu/gorm.(*DB).Save(0xc20806c840, 0x432da80, 0xc2080b6210, 0xc20803b350)
	/Users/ikuo/lib/go/src/github.com/jinzhu/gorm/main.go:267 +0x96
main.main()
	/Users/ikuo/sandbox/psql/main.go:41 +0x782

goroutine 5 [chan receive]:
database/sql.(*DB).connectionOpener(0xc208046000)
	/usr/local/Cellar/go/1.4.2/libexec/src/database/sql/sql.go:589 +0x4c
created by database/sql.Open
	/usr/local/Cellar/go/1.4.2/libexec/src/database/sql/sql.go:452 +0x31c

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/Cellar/go/1.4.2/libexec/src/runtime/asm_amd64.s:2232 +0x1
exit status 2
```

--- a/postgres.go
+++ b/postgres.go
@@ -106,7 +106,12 @@ func (h Hstore) Value() (driver.Value, error) {
 	}
 
 	for key, value := range h {
-		hstore.Map[key] = sql.NullString{String: *value, Valid: true}
+		var s sql.NullString
+		if value != nil {
+			s.String = *value
+			s.Valid = true
+		}
+		hstore.Map[key] = s
 	}
 	return hstore.Value()
 }

